### ISSUE TITLE
Plugin doesn't respect Brunch conventions configuration

### ISSUE THREAD

**Issue by TomRK1089** [brunch/typescript-brunch#37]
The plugin doesn't properly delegate to Brunch's [conventions](http://brunch.io/docs/config#-conventions-) property on config. There is a hard-coded regex against `(^bower_components|vendor|node_modules)` at https://github.com/brunch/typescript-brunch/blob/6b6b114d7314285d4a244ef45a6a007fc8bae37d/index.js#L81

This means that if you have code in your app/ directory under a directory that starts with "vendor", even if you've properly adjusted the Brunch conventions, the Typescript files won't be compiled. On the latest versions of both Brunch and typescript-brunch, this causes the original Typescript source to be passed to deppack, resulting in a very confusing error when a statement of the form
```
import Foo = require("bar");
```
is found.

This plugin should create the matcher by using the value of conventions.vendor rather than hard-coding a constant value. Note that the conventions properties may themselves be functions or regexes rather than string literals, e.g.
```
conventions: {
  vendor: function() { return false; }
}
```

**Comment by shvaikalesh** [comment]
Hey @TomRK1089, thanks for the report. Indeed, plugins should try to infer settings from config as much as possible. I will gladly review a PR.

**Comment by TomRK1089** [comment]
@shvaikalesh I'd like to put one together, but I'm not familiar with testing/debugging Brunch plugins. If there's a README or other getting started guide that can help me validate the fix I'd be happy to submit a PR.

**Comment by TomRK1089** [comment]
I managed to figure it out based on the Brunch manual and have opened PR #38 to track this issue. Please let me know if there's anything else I can provide.

**Comment by shvaikalesh** [comment]
Closed by #38
